src/routes/fetchTree.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import axios from "axios";
+import { getUrlToFetch } from "../utils/fetchData";
+import LandingPageRoute from "./landingPageRoute";
+import { ACTION_TYPES } from "../reducers/actionTypes";
+import { ingestNextstrain } from "../utils/nextstrainAdapter";
+
+/**
+ * Handles fetching external tree JSON and loading it into app.
+ *
+ * Necessary to have this as a separate function because `async`. Since the
+ * result of network call happens async and we need to `await` that, we must
+ * do it in this separate async function to enable us to block on the `await`.
+ *
+ * If there's an issue with fetching the external data, it kicks off an error
+ * action to redux so we can display an error to the user. Sadly, we can't
+ * really display very concrete stuff on why there was an error. For example,
+ * CORS errors are totally opaque to us
+ *   (see https://github.com/axios/axios/issues/838)
+ * so we pretty much just need to tell the user something generic and leave
+ * it up to them to troubleshoot. Hopefully over time we'll get a better
+ * sense of what trips up our users and we can write a useful help guide.
+ *
+ * NOTE: if we eventually handle anything other than nextstrain tree json, we
+ * need a way to know which json type we're trying to ingest. Maybe we could
+ * determine it from structure of json and auto-choose?
+ */
+async function handleDataFetch(targetUrl: string, dispatch: Function) {
+  let response; // Here because of `try` block scope.
+  try {
+    response = await axios.get(targetUrl);
+  } catch {
+    const errorMessage = `Attempted to fetch data from URL: ${targetUrl}, but `
+      + "there was a problem with fetching the data. Please confirm "
+      + "the URL is publicly accessible and has the appropriate CORS policy "
+      + "in place. You may want to look in your browser console."
+    dispatch({
+      type: ACTION_TYPES.FETCH_TREE_DATA_FAILED,
+      errorMessage,
+    });
+    return; // Can't progress since fetch failed.
+  }
+
+  // TODO Should have some kind of sanity check that we got a Nextstrain tree.
+  // If it fails sanity check, kick off an error message about malformed.
+  const ingestedNextstrain = ingestNextstrain(response.data);
```

Comment:
Making a note to myself to add a conditional here along the lines of 
```if (ingestedNextstrain) {do what it currently does} else {dispatch({type: found url but json was bad, data: error message}```
and do error checking within `ingestnextstrain`

Reply:
Exactly! I'm not sure what the best approach is. I see three:
- What you're talking about, where `ingestNextstrain` returns `null` or something when ingestion goes badly. (implicit meaning in the return value)
- Having `ingestNextstrain` raise an error (`throw`) when ingestion goes badly, and catching it with a try/catch instead of if/else at this level. (ask forgiveness not permission model)
- Having a helper function that first verifies if object is acceptable, `checkIfLegalNextstrain` or whatever. This might help if we eventually start supporting lots of tree formats, because we might some way to check for the type of tree automatically, then `ingest` accordingly. (look before you leap model)

All of the above are reasonable approaches! Just want to present options to think about.

Reply:
Ahh, yes. The helper function is a lot cleaner if / when we accept multiple formats. Thanks for the food for thought!